snap/hooks/post-refresh

```diff
@@ -1,5 +1,9 @@
 #!/bin/bash -e
 
+# ensure the layouts dir for /etc/{stuff} exists
+mkdir -p "$SNAP_DATA/etc/docker"
```

Comment:
In theory we shouldn't need to make sure that the `/etc/docker` directory exists, since it is created in the install hook for many years (since 2018), so I bet that all current installations should have that in place. On the other side, this is harmless.

Reply:
I agree it very likely not required for most scenarios in post-refresh, especially as it's been in place in inst for while.

Really, I just did to sync between the two hooks, because I think it makes more sense to just sync that rather than try and maintain it over time.  e.g.  I added the `mkdir` for `nvidia-container-runtime` in install and post-refresh because it was new, and then after _some_ arbitrary period you could argue that it could be removed from post-refresh, but perhaps it's just simpler to leave it synced.

I think it's ok, but I'm happy to remove it if you prefer.

Reply:
I would be fine in adding this to post-refresh.

Reply:
Cool - let's leave then :+1: